src/mpl/src/shm/mpl_shm.c

```diff
@@ -34,8 +34,14 @@ int MPL_shm_hnd_deserialize(MPL_shm_hnd_t hnd, const char *str_hnd, size_t str_h
     int rc = MPL_SHM_SUCCESS;
     MPLI_shm_hnd_reset_val(hnd);
     rc = MPLI_shm_ghnd_alloc(hnd, MPL_MEM_SHM);
+    if (rc != MPL_SHM_SUCCESS)
+        goto fn_exit;
```

Comment:
I personally prefer `return rc;` directly, but I am OK with this.

Reply:
I think I like the early return version better, too. I'll resubmit along with typo fixes.